File: Exemplos/App_eXperience_Flutter/Elgin_AppExperience_Flutter-font/android/app/src/main/java/com/elgin/flutter_m8/Paygo.java
package com.elgin.flutter_m8;

import android.annotation.SuppressLint;
import android.app.Activity;
import android.content.pm.PackageManager;
import android.os.Handler;
import android.widget.Toast;

import com.elgin.flutter_m8.TefEnums.FormaFinanciamento;
import com.elgin.flutter_m8.TefEnums.FormaPagamento;

import java.io.File;
import java.util.Map;

import org.json.JSONException;
import org.json.JSONObject;

import br.com.setis.interfaceautomacao.Cartoes;
import br.com.setis.interfaceautomacao.Confirmacoes;
import br.com.setis.interfaceautomacao.DadosAutomacao;
import br.com.setis.interfaceautomacao.Financiamentos;
import br.com.setis.interfaceautomacao.ModalidadesPagamento;
import br.com.setis.interfaceautomacao.Operacoes;
import br.com.setis.interfaceautomacao.EntradaTransacao;
import br.com.setis.interfaceautomacao.Personalizacao;
import br.com.setis.interfaceautomacao.SaidaTransacao;
import br.com.setis.interfaceautomacao.StatusTransacao;
import br.com.setis.interfaceautomacao.Transacoes;
import br.com.setis.interfaceautomacao.Versoes;
import br.com.setis.interfaceautomacao.ViasImpressao;

public class Paygo {
    private Activity mainActivity;

    private SaidaTransacao mSaidaTransacao;
    private DadosAutomacao mDadosAutomacao;
    private Transacoes mTransacoes;
    private Confirmacoes mConfirmacao;
    private EntradaTransacao mEntradaTransacao = null;
    private Personalizacao mPersonalizacao;
    private Versoes versoes;
    private Handler mHandler;

    public Paygo(Activity activity) {
        mainActivity = activity;
        IniciaClassesInterface(false, false, false, false);
        mHandler = new Handler();
    }

    private final Runnable mostraJanelaResultado = new Runnable() {//region
        public void run() {
            JSONObject dataJson = new JSONObject();

            String retorno = mSaidaTransacao.obtemMensagemResultado();
            String via_cliente = "";


            if (mSaidaTransacao.obtemInformacaoConfirmacao()) {
                mConfirmacao.informaStatusTransacao(StatusTransacao.CONFIRMADO_AUTOMATICO);
                mTransacoes.confirmaTransacao(mConfirmacao);

                ViasImpressao vias = mSaidaTransacao.obtemViasImprimir();
                System.out.println("VIAS: " + vias.equals("VIA_NENHUMA"));

                //Imprime a via do cliente
                if (vias == ViasImpressao.VIA_CLIENTE || vias == ViasImpressao.VIA_CLIENTE_E_ESTABELECIMENTO) {
                    via_cliente = mSaidaTransacao.obtemComprovanteGraficoPortador();
                }

                try {
                    dataJson.put("retorno", retorno);
                    dataJson.put("via_cliente", via_cliente);
                } catch (JSONException e) {
                    e.printStackTrace();
                }
                MainActivity.resultFlutter.success(dataJson.toString());
            }
        }
    };

//    public void mostraResultadoTransacao(){
//        String retorno = "";
//        retorno = mSaidaTransacao.obtemMensagemResultado();
//
//        if(mSaidaTransacao.obtemInformacaoConfirmacao()) {
//            mConfirmacao.informaStatusTransacao(StatusTransacao.CONFIRMADO_AUTOMATICO);
//            mTransacoes.confirmaTransacao(mConfirmacao);
//
//            ViasImpressao vias = mSaidaTransacao.obtemViasImprimir();
//            System.out.println("VIAS: " + vias.equals("VIA_NENHUMA"));
//
//            //Imprime a via do cliente
//            if ( vias == ViasImpressao.VIA_CLIENTE || vias == ViasImpressao.VIA_CLIENTE_E_ESTABELECIMENTO) {
//                String via_cliente = mSaidaTransacao.obtemComprovanteGraficoPortador();
//                retorno = via_cliente;
//            }
//
//            MainActivity.resultFlutter.success(retorno);
//        }
//    }


    public void efetuaTransacao(Operacoes operacao, Map<String, Object> params) {
        mEntradaTransacao = new EntradaTransacao(operacao, "1");

        if (!operacao.equals(Operacoes.ADMINISTRATIVA)) {
            final String valor = (String) params.get("valor");
            final int parcelas = (Integer) params.get("parcelas");
            final FormaPagamento formaPagamentoSelecionada = FormaPagamento.fromRotulo((String) params.get("formaPagamento"));
            final FormaFinanciamento formaFinanciamentoSelecionada = FormaFinanciamento.fromRotulo((String) params.get("formaFinanciamento"));

            mEntradaTransacao.informaValorTotal(valor);

            if (operacao.equals(Operacoes.VENDA)) {
                mEntradaTransacao.informaDocumentoFiscal("1000");
            }

            mEntradaTransacao.informaModalidadePagamento(ModalidadesPagamento.PAGAMENTO_CARTAO);

            switch (formaPagamentoSelecionada) {
                case CREDITO:
                    mEntradaTransacao.informaTipoCartao(Cartoes.CARTAO_CREDITO);
                    mEntradaTransacao.informaNumeroParcelas(parcelas);
                    break;
                case DEBITO:
                    mEntradaTransacao.informaTipoCartao(Cartoes.CARTAO_DEBITO);
                    mEntradaTransacao.informaNumeroParcelas(parcelas);
                    break;
                case TODOS:
                    mEntradaTransacao.informaTipoCartao(Cartoes.CARTAO_DESCONHECIDO);
                    break;
            }

            switch (formaFinanciamentoSelecionada) {
                case LOJA:
                    mEntradaTransacao.informaTipoFinanciamento(Financiamentos.PARCELADO_ESTABELECIMENTO);
                    break;
                case ADM:
                    mEntradaTransacao.informaTipoFinanciamento(Financiamentos.PARCELADO_EMISSOR);
                    break;
                case A_VISTA:
                    mEntradaTransacao.informaTipoFinanciamento(Financiamentos.A_VISTA);
                    break;
            }

            mEntradaTransacao.informaNomeProvedor("DEMO");
            mEntradaTransacao.informaCodigoMoeda("986");
        }

        mConfirmacao = new Confirmacoes();

        new Thread(() -> {
            try {
                mDadosAutomacao.obtemPersonalizacaoCliente();
                mSaidaTransacao = mTransacoes.realizaTransacao(mEntradaTransacao);

                if (mSaidaTransacao == null)
                    return;

                mConfirmacao.informaIdentificadorConfirmacaoTransacao(
                        mSaidaTransacao.obtemIdentificadorConfirmacaoTransacao()
                );

                mEntradaTransacao = null;
            } catch (Exception e) {
                System.out.println("Exception" + e);
            } finally {
                mEntradaTransacao = null;
                if (mSaidaTransacao != null) {
                    mHandler.post(mostraJanelaResultado);
                }
            }
        }).start();
    }

    private void IniciaClassesInterface(boolean suportaViasDiferenciadas, boolean suportaViasReduzidas, boolean troco, boolean desconto) {
        String versaoAutomacao;
        try {
            versaoAutomacao = mainActivity.getPackageManager().getPackageInfo(
                    mainActivity.getPackageName(), 0).versionName;
        } catch (PackageManager.NameNotFoundException e) {
            versaoAutomacao = "Indisponivel";
        }

        mPersonalizacao = setPersonalizacao(false);

        mDadosAutomacao = new DadosAutomacao("Automacao Demo", versaoAutomacao, "SETIS",
                troco, desconto, suportaViasDiferenciadas, suportaViasReduzidas, true, mPersonalizacao);

        mTransacoes = Transacoes.obtemInstancia(mDadosAutomacao, mainActivity);
        versoes = mTransacoes.obtemVersoes();
    }

    private Personalizacao setPersonalizacao(boolean isInverse) {
        Personalizacao.Builder pb = new Personalizacao.Builder();

        try {
            if (isInverse) {
                pb.informaCorFonte("#000000");
                pb.informaCorFonteTeclado("#000000");
                pb.informaCorFundoCaixaEdicao("#FFFFFF");
                pb.informaCorFundoTela("#F4F4F4");
                pb.informaCorFundoTeclado("#F4F4F4");
                pb.informaCorFundoToolbar("#2F67F4");
                pb.informaCorTextoCaixaEdicao("#000000");
                pb.informaCorTeclaPressionadaTeclado("#e1e1e1");
                pb.informaCorTeclaLiberadaTeclado("#dedede");
                pb.informaCorSeparadorMenu("#2F67F4");
            }
        } catch (IllegalArgumentException e) {
            Toast.makeText(mainActivity, "Verifique valores de\nconfiguração", Toast.LENGTH_SHORT).show();
        }
        return pb.build();
    }
}

File: Exemplos/App_eXperience_XamarinAndroid/Elgin_AppExperience_XamarinAndroid/Tef/PayGo.cs
using Android.App;
using Android.Content;
using Android.OS;
using Android.Runtime;
using Android.Views;
using Android.Widget;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

using BR.Com.Setis.Interfaceautomacao;
using Android.Content.PM;

using System.Threading;
using static M8.Tef;

namespace M8
{
    class PayGo
    {
        private static Activity mainActivity;

        private static SaidaTransacao mSaidaTransacao;
        private static DadosAutomacao mDadosAutomacao;
        private static Transacoes mTransacoes;
        private static Confirmacoes mConfirmacao;
        private static EntradaTransacao mEntradaTransacao = null;
        private static Personalizacao mPersonalizacao;
        private static Versoes versoes;
        private static Handler mHandler;

       

        public PayGo()
        {
        }

        public static void setActivity(Activity activity)
        {
            mainActivity = activity;
            IniciaClassesInterface(false, false, false, false);
            #pragma warning disable CS0618 // O tipo ou membro é obsoleto
            mHandler = new Handler();
        }

        private static readonly Java.Lang.Runnable mostraJanelaResultado = new Java.Lang.Runnable(new Action(delegate
        {
            Dictionary<string, string> mapValues = new Dictionary<string, string>();

            string retorno = mSaidaTransacao.ObtemMensagemResultado();
            string via_cliente = "";


            if (mSaidaTransacao.ObtemInformacaoConfirmacao())
            {
                mConfirmacao.InformaStatusTransacao(StatusTransacao.ConfirmadoAutomatico);
                mTransacoes.ConfirmaTransacao(mConfirmacao);



                ViasImpressao vias = mSaidaTransacao.ObtemViasImprimir();
                Console.WriteLine("VIAS: " + vias.Equals("VIA_NENHUMA"));

                //Imprime a via do cliente
                if (vias == ViasImpressao.ViaCliente || vias == ViasImpressao.ViaClienteEEstabelecimento)
                {
                    via_cliente = mSaidaTransacao.ObtemComprovanteGraficoPortador();
                }

                mapValues.Add("retorno", retorno);
                mapValues.Add("via_cliente", via_cliente);



                Tef.OptionsReturnPaygo(mapValues);
            }
        }));

        public void mostraResultadoTransacao()
        {
            string retorno = mSaidaTransacao.ObtemMensagemResultado();

            if (mSaidaTransacao.ObtemInformacaoConfirmacao())
            {
                mConfirmacao.InformaStatusTransacao(StatusTransacao.ConfirmadoAutomatico);
                mTransacoes.ConfirmaTransacao(mConfirmacao);

                ViasImpressao vias = mSaidaTransacao.ObtemViasImprimir();
                Console.WriteLine("VIAS: " + vias.Equals("VIA_NENHUMA"));

                //Imprime a via do cliente
                if (vias == ViasImpressao.ViaCliente || vias == ViasImpressao.ViaClienteEEstabelecimento)
                {
                    string via_cliente = mSaidaTransacao.ObtemComprovanteGraficoPortador();
                    retorno = via_cliente;
                }
            }
        }





        public static void efetuaTransacao(Operacoes operacao, Dictionary<string, object> dictionary)
        {
            mEntradaTransacao = new EntradaTransacao(operacao, "1");
            
            if (!operacao.Equals(Operacoes.Administrativa))
            {
                string valor = (string)dictionary["valor"];
                int parcelas = int.Parse((string)dictionary["parcelas"]);
                FormaPagamento formaPagamento = (FormaPagamento)dictionary["formaPagamento"];
                FormaFinanciamento tipoParcelamento = (FormaFinanciamento)dictionary["tipoParcelamento"];

                mEntradaTransacao.InformaValorTotal(valor);

                if (operacao.Equals(Operacoes.Venda))
                {
                    mEntradaTransacao.InformaDocumentoFiscal("1000");
                }

                mEntradaTransacao.InformaModalidadePagamento(ModalidadesPagamento.PagamentoCartao);

                switch (formaPagamento)
                {
                    case FormaPagamento.CREDITO:
                        mEntradaTransacao.InformaTipoCartao(Cartoes.CartaoCredito);
                        mEntradaTransacao.InformaNumeroParcelas(parcelas);
                        break;
                    case FormaPagamento.DEBITO:
                        mEntradaTransacao.InformaTipoCartao(Cartoes.CartaoDebito);
                        mEntradaTransacao.InformaNumeroParcelas(parcelas);
                        break;
                    case FormaPagamento.TODOS:
                        mEntradaTransacao.InformaTipoCartao(Cartoes.CartaoDesconhecido);
                        break;
                }

                switch (tipoParcelamento)
                {
                    case FormaFinanciamento.LOJA:
                        mEntradaTransacao.InformaTipoFinanciamento(Financiamentos.ParceladoEstabelecimento);
                        break;
                    case FormaFinanciamento.ADM:
                        mEntradaTransacao.InformaTipoFinanciamento(Financiamentos.ParceladoEmissor);
                        break;
                    case FormaFinanciamento.A_VISTA:
                        mEntradaTransacao.InformaTipoFinanciamento(Financiamentos.AVista);
                        break;
                }

                mEntradaTransacao.InformaNomeProvedor("DEMO");
                mEntradaTransacao.InformaCodigoMoeda("986");
            }

            mConfirmacao = new Confirmacoes();
            
            new Thread( ()=> {
            try
            {
                mDadosAutomacao.ObtemPersonalizacaoCliente();
                mSaidaTransacao = mTransacoes.RealizaTransacao(mEntradaTransacao);

                if (mSaidaTransacao == null)
                    return;

                mConfirmacao.InformaIdentificadorConfirmacaoTransacao(
                        mSaidaTransacao.ObtemIdentificadorConfirmacaoTransacao()
                );

                mEntradaTransacao = null;
            }
            catch (Exception e)
            {
                Console.WriteLine("Exception" + e);
            }
            finally
            {
                mEntradaTransacao = null;
                if (mSaidaTransacao != null)
                {
                    mHandler.Post(mostraJanelaResultado);
                }
          
          }
        }).Start();
        }





        private static void IniciaClassesInterface(bool suportaViasDiferenciadas, bool suportaViasReduzidas, bool troco, bool desconto)
        {
            string versaoAutomacao;
            try
            {
                //versaoAutomacao = mainActivity.getPackageManager().getPackageInfo(
                       // mainActivity.getPackageName(), 0).versionName;
                versaoAutomacao = mainActivity.PackageManager.GetPackageInfo(mainActivity.PackageName, 0).VersionName;
            }
            catch (PackageManager.NameNotFoundException e)
            {
                Console.WriteLine("Exception" + e);
                versaoAutomacao = "Indisponivel";
            }

            mPersonalizacao = setPersonalizacao(false);

            mDadosAutomacao = new DadosAutomacao("Automacao Demo", versaoAutomacao, "SETIS",
                    troco, desconto, suportaViasDiferenciadas, suportaViasReduzidas, true, mPersonalizacao);

            mTransacoes = Transacoes.ObtemInstancia(mDadosAutomacao, mainActivity);
            versoes = mTransacoes.ObtemVersoes();
        }
        private static Personalizacao setPersonalizacao(bool isInverse)
        {
            Personalizacao.Builder pb = new Personalizacao.Builder();
            
            try
            {
                if (isInverse)
                {
                    pb.InformaCorFonte("#000000");
                    pb.InformaCorFonteTeclado("#000000");
                    pb.InformaCorFundoCaixaEdicao("#FFFFFF");
                    pb.InformaCorFundoTela("#F4F4F4");
                    pb.InformaCorFundoTeclado("#F4F4F4");
                    pb.InformaCorFundoToolbar("#2F67F4");
                    pb.InformaCorTextoCaixaEdicao("#000000");
                    pb.InformaCorTeclaPressionadaTeclado("#e1e1e1");
                    pb.InformaCorTeclaLiberadaTeclado("#dedede");
                    pb.InformaCorSeparadorMenu("#2F67F4");
                }
            }
            catch (Java.Lang.IllegalArgumentException)
            {
                Toast.MakeText(mainActivity, "Verifique valores de\nconfiguração", ToastLength.Short).Show();
            }
            return pb.Build();
        }
    }
}